### ISSUE TITLE
[機能] Court Detectionに6DoF query encoderと軽量decoderアブレーション基盤を追加する

### ISSUE THREAD

**Issue by Motoki0705** [Motoki0705/tennis-lab#779]
## 概要

Court Detection に、`DINOv3 patch tokens → task Transformer encoder → dense decoder` の新規 model variant を追加する。task Transformer には learned pose query を1個加え、最終 query tokenからcameraのtarget-court canonical poseとfocal scalarを10Dで回帰し、patch token側からtarget-court KP14を中心とするdense targetを生成する。

```text
RGB
 └─ DINOv3 patch tokens + patch grid
     └─ projection + 2D RoPE + learned pose query（queryは位置encodingなし）
         └─ task Transformer encoder（MHA）
             ├─ final pose query ── MLP ── [tx,ty,tz,a11,a12,a13,a21,a22,a23,logf]
             └─ patch-token taps ── dense decoder ── kp / seg / line heads
```

仮説は、幾何推論をtask Transformer encoderに集約できるなら、dense decoderは機械的な空間復元に近い小型構成で十分である、というもの。まずencoder depthを主軸とするscaling lawを推定し、その後、選定encoderを固定してKP decoderのfamily/size scalingを測る。最後にposeと複数dense教師の組合せを比較し、内部表現への寄与を分離する。

## 確定する座標・出力contract

- #778により対象courtをsample-level target courtへ一意化し、#782のSynthetic Court V3 camera-end canonicalizationを正本として使用する。legacy V2 KP14をcorrected V3としてsilentに受理しない。
- target authorityは `T_scene_from_camera = camera.camera_to_scene` と `T_scene_from_court = target_court.binding.scene_from_court`。まず `T_court_from_camera = inverse(T_scene_from_court) @ T_scene_from_camera` を導出し、#782と同じ `S=I|Rz(pi)` を適用して `T_canonical_from_camera = T_canonical_from_court @ T_court_from_camera` を得る。
- translation `[tx,ty,tz]` はtarget-court中心を原点とするcanonical court座標[m]でのcamera center。
- rotationはOpenCV camera軸からcanonical court軸へ写す `R_canonical_from_camera ∈ SO(3)`。headはその先頭2行をrow-majorで `[a11,a12,a13,a21,a22,a23]` として出力し、decode時にrow-wise Gram–Schmidtとcross productでproper rotationを復元する。退化・nonfinite入力をsilent normalizeしない。
- focalはaugmentation後画像のpixel単位で `fx=fy=f>0` を要求し、10番目の値を `logf=log(f)` とする。decode後はfiniteかつpositiveでなければならない。
- raw 10D、decoded pose/intrinsics、loss input、prediction persistenceを型付きcontractで分け、tuple位置や暗黙dict keyに依存しない。

### 1. 新規model variantとHydra軸

- 既存 `court_hierarchical` と既存 `model/encoder=dinov3 model/decoder=dpt` のdefault contractは変更しない。
- `src/tasks/court_detection/models/` に新variantを追加し、責務を `backbone / task_encoder / decoder / heads` に分ける。
  - `model/backbone`: DINOv3 asset、出力layer、`frozen / last_n / full / LoRA`
  - `model/task_encoder`: `hidden_dim`, `depth`, `num_heads`, `mlp_ratio`, dropout, 2D RoPE、`tap_indices`
  - `model/decoder`: family、width、stage/fusion level、tap、upsample/reassemble設定
  - `model/heads`: pose10d MLP容量と非空のdense target subset
- `tiny / small / base` presetとraw overrideを用意し、head divisibility、RoPE head dimension、depth/tap、grid/token、target/loss、未知familyの不整合はtyped configで副作用前に拒否する。
- DINOv3自身のCLS/register tokenは初期比較で使用せず、learned pose queryの効果と混同しない。

### 2. Task Transformer encoder scaling

- DINOv3 boundaryはpatch token `[B,N,C]` と実patch grid `(Hp,Wp)` を返し、padding後gridとtoken数をstrictに検証する。
- patch tokenのMHA Q/Kに2D RoPEを適用する。pose queryは位置を持たず、query Q/KにはRoPEを適用しない。queryとpatchは同じself-attentionで相互作用する。
- encoder tapはqueryを除いたpatch tokenとgridを型付きで返す。tap index重複、範囲外、shape不一致をrejectする。
- 最初の実験はdecoder/headとDINOv3条件を固定し、主にencoder depthを増やしたときのpose/KP精度、parameter/MACs、latencyの変化を15 epoch、約256pxの同一入力条件、3 seed以上で図示する。width等の追加軸はdepth則を読める範囲に限定する。

### 3. Pose10D教師・loss・metric

- camera/court authorityから10D targetを一意に導出し、既知3D court点のprojection round-tripでtranslation、rotation、focalの規約を検証する。
- rotation lossはdecoded SO(3)のgeodesic、translation lossはmetric-space regression、focal lossはlog-focal regressionとし、各lossとweighted totalを個別logする。
- metricは最低限translation L2 [m]、rotation geodesic [deg]、focal relative error、`|logf_pred-logf_gt|` を記録する。
- pose targetを要求しながらV3 target-court camera authorityがないsourceを選んだ場合、DataLoader worker/GPU初期化前に失敗させる。

### 4. `fx=fy`を壊さないaugmentation option

- configから `preserve_fx_fy=true|false` を選べるようにする。trueではisotropic resize/scaleと明示的なpadding/crop calibration更新だけを許可し、anisotropic resize/scaleや未対応geometric transformをcompose時にrejectする。
- 比較実験はtrueを使用し、約256px canvasへisotropic resize + paddingする。更新後Kで `fx=fy`、principal-point変換、既知3D点projection round-tripを検証する。
- horizontal flip等のposeを変えるaugmentationはcamera/canonical frameとKを同時に変換できる場合だけ許可し、未対応ならpose supervisionとの組合せを明示的にrejectする。

### 5. Singleton KP14とdecoder scaling

- #778 target-court scopeと#782 V3により、KP14 heatmapは各channelちょうど1点のtargetとする。各channelのpeak/mean-distance metricもsingleton identityで計算し、multi-peak最短距離へfallbackしない。
- 共通dense output contractは入力画像解像度のfeature mapとし、選択headを適用する。
  1. `linear`: final patch grid projection + fixed resizeのみ。token mixingなし。
  2. `progressive`: configurableな少数upsample + depthwise-separable convolution block。
  3. `dpt`: encoder複数tapのreassemble/fuse。widthとfusion levelを独立可変。
- encoder scalingから選んだ構成を固定し、KPを評価targetとしてdecoder family/sizeを15 epoch、約256px、3 seed以上で比較する。decoder parameters/MACs/latency/peak memoryとKP singleton metricを同じ図表にする。

### 6. Multimodal supervisionと実験順序

1. 固定decoder/headでencoder depth scalingを推定する。
2. 選定encoderを固定し、singleton KP14でdecoder scalingを推定する。
3. 同一architectureでpose supervision on/off、`kp / seg / line` の非空subsetを比較し、異種教師信号の寄与を分離する。
- epochは各run 15、入力はisotropicな約256px条件、split/checkpoint/backbone train mode/augmentation/loss weightは比較内で固定する。
- 各presetでdecoder parameters、trainable/total parameters、decoder MACs、decoder単体latency、end-to-end latency、peak memoryを同一shapeで記録する。
- encoder/decoder scaling curve、3-seed平均・分散、精度・容量・速度Paretoを作成し、全runとgroup findingsを`knowledge/`へ登記する。

## スコープ

- 対象: 新規model variant、DINOv3 token boundary、MHA+2D RoPE task encoder、positionless pose query、pose10d head、軽量decoder family、strict typed config/Hydra composition。
- 対象: Synthetic Court V3 target-court canonical pose/focal target、augmentation-aware K、loss/metric/checkpoint/persistence contract。
- 対象: target-court singleton KP14、`kp / seg / line` 非空subset、loss weight別logging、CPU unit/integration test、容量profile、再現可能ablation config、GPU比較実験、`knowledge/`登記。
- 対象外: real datasetへのpose擬似教師、時系列/multi-view化、DINOv3 SSL recipe再探索、自動loss balancing、根拠なしのproduction default変更。
- `seg / line` は別Issueでsingle-court化されない限り現行all-court semanticsを維持し、結果上もKPとの意味の違いを明記する。

## Acceptance checklist

- [ ] 既存 `court_hierarchical` defaultと既存DINOv3+DPT baselineが変更前と同じcontractでcompose・forwardできる。
- [ ] 新variantで `backbone / task_encoder / decoder / heads` を独立Hydra overrideでき、preset/raw値と不整合をstrict typed configで検証できる。
- [ ] DINOv3 patch token/grid/padding、positionless pose query 1個、patch-only 2D RoPE、encoder tap shapeが明示contractになり、不正shape/tapをsilent reshapeしない。
- [ ] pose query→MLPのraw出力が正確に `[tx,ty,tz,a11,a12,a13,a21,a22,a23,logf]` の `[B,10]` である。
- [ ] decoded rotationがfiniteなproper SO(3)、translationがtarget-court canonical metres、focalがaugmentation後pixel単位のfinite positive scalarである。
- [ ] Synthetic Court V3のcamera/target-court authorityと#782の `S=I|Rz(pi)` からpose10dを一意に導出し、既知3D court点のprojection round-tripが成立する。
- [ ] `preserve_fx_fy` optionがHydra/typed configから選べ、trueではaugmentation後 `fx=fy` とK/pose整合を保証し、非対応transformを副作用前にrejectする。
- [ ] pose非対応source、authority欠落、unsupported augmentation、target/loss矛盾をDataLoader worker/GPU初期化前に明示エラーにする。
- [ ] `linear / progressive / dpt` が共通dense contractを満たし、width/stage/fusion/tapを変えても選択head出力が入力画像解像度と一致する。
- [ ] V3 target-court KP14 heatmapは各channelちょうど1点で、metricもsingleton identityを評価し、multi-peak fallbackしない。
- [ ] pose translation/rotation/focal、各dense loss、weighted totalが個別logされ、supervision subsetがconfig名とcheckpoint schemaから識別できる。
- [ ] lightweight shape/transform/loss/config testsは`tests/unit/tasks/court_detection/`でsrc mirrorし、Hydra composeと最小forward/loss/backwardは`tests/integration/tasks/court_detection/`のCPU smokeで検証する。
- [ ] fake DINOv3でquery/patch tap、全decoder family、10D head、finite loss、backpropを実weightなしで検証し、関連pytest・ruff・mypyがPASSする。
- [ ] encoder depth sweepを先に、選定encoderでKP decoder sweepを次に、pose/dense supervision sweepを最後に、各15 epoch・約256px・3 seed以上で実行する。
- [ ] encoder/decoder scaling curveと精度・分散・容量・速度Paretoを作成し、全run/group findings/採否根拠を`knowledge/`へ登記する。
- [ ] 実験採用まで既存production defaultを変更せず、不採用経路は明確なablation config/moduleとして隔離するか削除する。

## 補足

- Dependency: #778（target-court-only KP）、#782（V3 canonical KP/pose）。
- query/head先行例: `src/tasks/plcs/models/plcs_track_query_model.py`、`src/tasks/slcs/models/slcs_model.py`。
- rotation変換の既存共通実装: `src/utils/geometry/rotation_conversions.py`。
- 本Issueのcamera poseはcamera-to-canonical-court transformであり、headはtranslation 3 + row-wise rotation 6D + log-focal 1の10出力。quaternion 7Dは使用しない。
